app/Http/Controllers/App/PostController.php

```diff
@@ -59,11 +59,59 @@ public function store(Request $request)
         return redirect()->route('app.post.index');
     }
 
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Inertia\Response
+     */
+    public function edit(Request $request, int $id)
+    {
+        $post = Post::with('team.challenges', 'team.currentChallenges', 'team.banner')->findOrFail($id);
```

Comment:
By changing the route to `Route::get('post/edit/{post}', ...);`, you should be able to take advantage of the [Route Model Binding](https://laravel.com/docs/8.x/routing#route-model-binding):
```suggestion
    public function edit(Request $request, Post $post)
    {
        $post = $post->load('team.challenges', 'team.currentChallenges', 'team.banner');
```